astropy/utils/iers/iers.py

```diff
@@ -443,5 +473,49 @@ def pm_source(self, i):
         """Set PM source flag for entries in IERS table"""
         return np.ones_like(i) * FROM_IERS_B
 
+class IERS_Auto(IERS_A):
```

Comment:
I did think about that.  My first issue is that the `IERS` class is already pretty confusing to me, so adding an option that completely changes the behavior (and in fact the *class* of the output) of `open()` is just too much.

The other problem that convinced me a new class is the best way is that all of the existing options to `IERS.open()` are moot in this case.  Thus you would need to raise an exception if any args or kwargs *besides* the `auto_update` option were provided.  That just seems like a bad API.